java/src/main/java/cucumber/runtime/java/AbstractJavaSnippet.java

```diff
@@ -24,7 +24,9 @@ public String arguments(List<Class<?>> argumentTypes) {
     public String tableHint() {
         return "    // For automatic transformation, change DataTable to one of\n" +
                 "    // List<YourType>, List<List<E>>, List<Map<K,V>> or Map<K,V>.\n" +
-                "    // E,K,V must be a scalar (String, Integer, Date, enum etc)\n";
+                "    // E,K,V must be a scalar (String, Integer, Date, enum etc)\n" +
+                "    // Make sure the variable names for YourType match the column names\n" +
+                "    // in your feature file.\n";
```

Comment:
I think 'The field names of YourType must match the column names' is better.

Field names instead of variables. The latter exists only in a function. The former is part of a type.

And must instead of make sure because of the rfc.

Not sure about the subject and object. Swapping might be better (if only someone told me learning grammar would allow me to discuss the use of language :) ).

Reply:
Updated. Hope this is better.